src/ring_theory/noetherian.lean

```diff
@@ -774,16 +774,31 @@ theorem is_noetherian_ring_of_ring_equiv (R) [comm_ring R] {S} [comm_ring S]
 is_noetherian_ring_of_surjective R S f.to_ring_hom f.to_equiv.surjective
 
 namespace submodule
+
+section map₂
+variables {R M N P : Type*}
+variables [comm_semiring R] [add_comm_monoid M] [add_comm_monoid N] [add_comm_monoid P]
+variables [module R M] [module R N] [module R P]
+
+theorem fg.map₂ (f : M →ₗ[R] N →ₗ[R] P) {p : submodule R M} {q : submodule R N}
+  (hp : p.fg) (hq : q.fg) : (map₂ f p q).fg :=
+let ⟨sm, hfm, hm⟩ := fg_def.1 hp, ⟨sn, hfn, hn⟩ := fg_def.1 hq in
+fg_def.2 ⟨set.image2 (λ m n, f m n) sm sn,
+  hfm.image2 _ hfn, map₂_span_span R f sm sn ▸ hm ▸ hn ▸ rfl⟩
```

Comment:
`fg_def.2 ⟨sm.image2 (λ m, f m) sn, hfm.image2 _ hfn, map₂_span_span R f sm sn ▸ hm ▸ hn ▸ rfl⟩`
golf it slightly and it fits in one line :)

Reply:
```
fg_def.2 ⟨sm.image2 f sn, hfm.image2 _ hfn, map₂_span_span R f sm sn ▸ hm ▸ hn⟩
```
even, no?

Reply:
`f` can't be coerced to a binary function without a lambda

Reply:
I'm not a huge fan of the asymmetry of `(λ m, f m)` aka `(λ m, ⇑(⇑f m))`; certainly I want to avoid that spelling in lemma statements, but I guess in a proof it doesn't really matter.

Reply:
Noted: don't use in statements.

Reply:
to @YaelDillies: the `rfl` is necessary and saves a `.symm` and parentheses.

Reply:
Note that the mystery triangles are just copied from the old proof, I haven't attempted to golf it further.